### PR TITLE
jshint cleanup

### PR DESCRIPTION
**Pull request by lewisje**
These changes are syntax changes recommended by jshint; I did not blindly trust the tool and frequently ignored its suggestions where I knew the tool didn't have the full story. All changes should result in code that runs the same way but is more formally correct; in the cases of moving function definitions outside loops so that new functions aren't created at each iteration, the code should even run faster, with less memory usage.
~~I had my Eyeo Contributor License Agreement here inline but decided I'd rather not let everyone on the Internet know my home address.~~

### CODE REVIEW

**Comment by palant** [comment]
I appreciate you making this effort, so far I only skimmed the changes. However, as with any automatic tool supposed to improve code quality, the results aren't very convincing. Most issues fixed here aren't something I would be concerned about in a review. For example, we had a discussion on `==` vs. `===` and `!=` vs. `!==` in JavaScript already - while strictly speaking the latter is better, its advantages are insignificant enough to ignore. On the other hand, changing code that has been tested always introduces the risk of regressions (like the undeclared variable noted above) which is why you normally don't do it unless you have a reason. So far the approach has been that we don't rewrite old code for the sake of rewriting. We can improve code if we change it anyway however.

Things get even more blurry when you look at `hasOwnProperty()` calls that have been added to a number of loops. We carefully avoid messing with `Object.prototype` exactly so we don't have to do it. Calling `hasOwnProperty()` in the loop makes things significantly slower, in addition to making the code harder to read.

To sum up: right now I am unsure about what should happen to your patches.

**Comment by fhd** [comment]
I mostly agree with @palant, although I'm definitely all for rewriting old code if we change our [coding style](https://adblockplus.org/en/coding-style) or decide to make the use of a linter like JSHint mandatory. (Rather than making changes as we go.)

But otherwise I agree. The majority of JSHint's suggestions are either trivial or questionable when it comes to our code. I'd rather not hurt consistency/readability without any real reasoning other than that JSHint said so. I'm generally not opposed to the use of a linter, I just think we'd have to make it only warn about potential problems we actually care about, and generally be in line with the coding style.

I didn't look into all the changes yet, but from what I can tell, these hints should generally be ignored:
- `===` vs `==` (the Mozilla coding style, which ours is based on, says to prefer `==`)
- `hasOwnProperty()` (what @palant said above)
- "Don't make functions within a loop" (it'd hurt readability without solving any real performance problems in all cases I've seen so far)

From what I've seen so far, the other changes are legitimate, missing semicolons and such.

**Comment by snoack** [comment]
I mostly a agree with @palant and @fhd. And since none of their comments have been addressed by @lewisje, and large parts of the changed code doesn't exist anymore, I'm going to close this pull request now.

**Comment by lewisje** [comment]
I'll just say that I agree with them too, and it was rather presumptuous of me to make wholesale changes to their coding style.

**Comment by palant** [comment]
As @fhd noted above, there are some valid points as well which are still present in the current code:
- Function `onHeadersReceived` in `webRequest.js` should be declared at the top scope, not inside an `if` block.
- Function `onMessage` in `options.js` shouldn't be followed by a semicolon.
- Assignments to `clickHideFiltersDialog.onmouseout` and `clickHideFiltersDialog.onmouseover` in `include.postload.js` are missing a semicolon.
- "array returend" in a comment in `include.postload.js` should be "array returned".
- Missing semicolon after `return locale` in `lib/utils.js` (then again, `Object.defineProperty` should be used instead of `__defineGetter__` there).
- Missing semicolon after `Services` declaration in `lib/compat.js`.

Maybe you could create a pull requests only containing these changes?
